### ISSUE TITLE
Ability to evaluate partial expressions in the REPL

### ISSUE THREAD

**Issue by be5invis** [idris-lang/Idris-dev#3768]
Currently REPL does not reduce partial expressions.
It is correct for typechecking, but for the REPL, there may should be an option to let it evaluate partials.

**Comment by ahmadsalim** [comment]
@be5invis Can you please provide an example? As far as I know it does reduce partial expressions.

**Comment by be5invis** [comment]
@luochen1990 paste the code that bothers you...

**Comment by be5invis** [comment]
@ahmadsalim 

@luochen1990 reports that he cannot successfully run the JSON parser sample of Lightyear.
this: https://github.com/ziman/lightyear/blob/master/examples/Json.idr
`parse jsonToplevelValue "[1, 2, 3]"` would not reduce to a `JsonValue`, stops at a gribrish.
(Perhaps the problem is not really about totality but laziness?)

**Comment by ahmadsalim** [comment]
@be5invis Ah, thanks.

**Comment by luochen1990** [comment]
Yes, I also post the question to reddit, and found that the [FAQ 2.5](http://docs.idris-lang.org/en/latest/faq/faq.html#evaluation-at-the-repl-doesn-t-behave-as-i-expect-what-s-going-on) explaned this problem, so I thaught it was expected...

But I do think that the REPL should eval the value instead of prints some IR code no matter it is total/partial or lazy/instant.

Thanks for all your help @be5invis @ahmadsalim :)

**Comment by ahmadsalim** [comment]
@be5invis @luochen1990  So I found out that if there is a lack of covering cases, then Idris will get stuck on the missing cases instead of providing an error at the REPL, which I agree is not very nice.
